internal/plugin/metadata_test.go

```diff
@@ -0,0 +1,137 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestValidatePluginData(t *testing.T) {
+
+	// A mock plugin with no commands
+	mockNoCommand := mockSubprocessCLIPlugin(t, "foo")
+	mockNoCommand.metadata.RuntimeConfig = &RuntimeConfigSubprocess{
+		PlatformCommands: []PlatformCommand{},
+		PlatformHooks:    map[string][]PlatformCommand{},
+	}
+
+	// A mock plugin with legacy commands
+	mockLegacyCommand := mockSubprocessCLIPlugin(t, "foo")
+	mockLegacyCommand.metadata.RuntimeConfig = &RuntimeConfigSubprocess{
+		PlatformCommands: []PlatformCommand{},
+		Command:          "echo \"mock plugin\"",
+		PlatformHooks:    map[string][]PlatformCommand{},
+		Hooks: map[string]string{
+			Install: "echo installing...",
+		},
+	}
+
+	// A mock plugin with a command also set
+	mockWithCommand := mockSubprocessCLIPlugin(t, "foo")
+	mockWithCommand.metadata.RuntimeConfig = &RuntimeConfigSubprocess{
+		PlatformCommands: []PlatformCommand{
+			{OperatingSystem: "linux", Architecture: "", Command: "sh", Args: []string{"-c", "echo \"mock plugin\""}},
+		},
+		Command: "echo \"mock plugin\"",
+	}
+
+	// A mock plugin with a hooks also set
+	mockWithHooks := mockSubprocessCLIPlugin(t, "foo")
+	mockWithHooks.metadata.RuntimeConfig = &RuntimeConfigSubprocess{
+		PlatformCommands: []PlatformCommand{
+			{OperatingSystem: "linux", Architecture: "", Command: "sh", Args: []string{"-c", "echo \"mock plugin\""}},
+		},
+		PlatformHooks: map[string][]PlatformCommand{
+			Install: {
+				{OperatingSystem: "linux", Architecture: "", Command: "sh", Args: []string{"-c", "echo \"installing...\""}},
+			},
+		},
+		Hooks: map[string]string{
+			Install: "echo installing...",
+		},
+	}
+
+	for i, item := range []struct {
+		pass bool
+		plug Plugin
+	}{
+		{true, mockSubprocessCLIPlugin(t, "abcdefghijklmnopqrstuvwxyz0123456789_-ABC")},
+		{true, mockSubprocessCLIPlugin(t, "foo-bar-FOO-BAR_1234")},
+		{false, mockSubprocessCLIPlugin(t, "foo -bar")},
+		{false, mockSubprocessCLIPlugin(t, "$foo -bar")}, // Test leading chars
+		{false, mockSubprocessCLIPlugin(t, "foo -bar ")}, // Test trailing chars
+		{false, mockSubprocessCLIPlugin(t, "foo\nbar")},  // Test newline
+		{true, mockNoCommand},     // Test no command metadata works
+		{true, mockLegacyCommand}, // Test legacy command metadata works
+		{false, mockWithCommand},  // Test platformCommand and command both set fails
+		{false, mockWithHooks},    // Test platformHooks and hooks both set fails
```

Comment:
You've done the work of describing the test, why not make that string part of the test data so the result can say which test failed (or succeeded)?